woocommerce/woocommerce-android
# Issue/11237 extract top performers view

Part of: #11237 


### Description

This PR adds the following changes: 
- Converts top performers' view to full composable code
- Adopts new implementation to handle all events 
- Extracts any top performers-related logic out of DashboardViewModel

This PR doesn't adjust the composable code to the recent designs published by Joe. Those came out after I had already worked on this. I will address that in a different PR. 

### Testing instructions

1. Enable `DYNAMIC_DASHBOARD` feature flag
2. Launch the app
3. From my store tab check top performers are loaded correctly and skeletons are shown
4. Click on a top performer item and check it opens the product detail
5. Load a custom range where there's not sales and check the empty case is displayed

### Images/gif



https://github.com/woocommerce/woocommerce-android/assets/2663464/0fe6b796-1580-407d-bef0-1d2d847ef666
